src/root_finding/krawczyk.jl

```diff
@@ -79,18 +79,20 @@ function krawczyk{T}(f::Function, f_prime::Function, x::Interval{T}, level::Int=
 
     debug && @show(x,m)
 
-    isthin(x) && return [(x, :unknown)]
+    isthin(x) && return [Root(x, :unknown)]
 
     # bisecting
     roots = vcat(
         krawczyk(f, f_prime, Interval(x.lo, m), level+1,
                  tolerance=tolerance, debug=debug, maxlevel=maxlevel),
-        krawczyk(f, f_prime, Interval(nextfloat(m), x.hi), level+1,
+        krawczyk(f, f_prime, Interval(m, x.hi), level+1,
```

Comment:
I think that the idea of adding `nextfloat` was a mistake. Conceptually, it is incorrect since it misses out the interval between the two floats (compare unums). From a practical point of view, the current version is better since it allows to check for a non-empty intersection to see if the two intervals are neighbours.